src/javascript/app/pages/form/currency-dropdown.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { CustomDropdown, DropdownItem, DropdownTitle, useDropdown } from '@deriv-com/quill-ui';
+
+const CurrencyContent = ({ currency_list, currency, onUpdate, elementId }) => {
+    const { close } = useDropdown();
+
+    return (
+        <div className='custom-dropdown-wrapper'>
+            {Object.keys(currency_list).map((key) => (
+                <div key={key}>
+                    <DropdownTitle label={key.toLocaleUpperCase()} />
+                    {currency_list[key].map((item) => (
+                        <DropdownItem
+                            key={item.value}
+                            label={item.text}
+                            selected={item.value === currency}
+                            onClick={() => {
+                                onUpdate(elementId, item.value, 'change');
+                                close();
+                            }}
+                        />
+                    ))}
+                </div>
+            ))}
+        </div>
+    );
+};
```

Comment:
perhaps, we don't need to separate `CurrencyContent` into a separate component above but instead leave it inside `CurrencyDropdown` since it's very small? you even decided to not separate it to another file :)
Same for `DropdownComponent`, `NumbersDropdown` and `TimePickerDropdown` below.

Reply:
For some reason the trigger to `close` doesn't work after refactor. Can we move this request as improvement after release?